src/org/thoughtcrime/securesms/notifications/NotificationCenter.java

```diff
@@ -570,17 +580,33 @@ public void removeAllNotifiations(int accountId) {
     public void updateVisibleChat(int accountId, int chatId) {
         Util.runOnAnyBackgroundThread(() -> {
 
-            visibleChat = new Pair<>(accountId, chatId);
             if (accountId != 0 && chatId != 0) {
+                visibleChat = new ChatData(accountId, chatId);
                 removeNotifications(accountId, chatId);
+            } else {
+                visibleChat = null;
             }
 
         });
     }
 
+    public void clearVisibleChat() {
+        visibleChat = null;
+    }
+
     public void maybePlaySendSound(DcChat dcChat) {
         if (Prefs.isInChatNotifications(context) && Prefs.isNotificationsEnabled(context) && !dcChat.isMuted()) {
             InChatSounds.getInstance(context).playSendSound();
         }
     }
+
+  private class ChatData {
```

Comment:
I think you need to override `equals`, otherwise `Util.equals` won't work

Reply:
done, thanks for pointing out, I forgot about it :sweat_smile: